Regression: no-identical-names false positive in sibling modules

When multiple QUnit modules are defined in the same file, they may have identical test names:

```js
QUnit.module('A');

QUnit.test('foo', assert => assert.ok(true));
QUnit.test('bar', assert => assert.ok(true));

QUnit.module('B');

QUnit.test('foo', assert => assert.ok(false));
QUnit.test('bar', assert => assert.ok(false));
```

However starting with eslint-plugin-qunit@5.2.0 `bar` is considered a duplicate:
```
foo.test.js
  9:12  error  Test name is used on line 4 in the same module  qunit/no-identical-names

✖ 1 problem (1 error, 0 warnings)
```

Interestingly `foo` is not marked as a duplicate, and in general it seems rather random: other combinations may produce the error.

This might have been introduced in #131.

--- a/lib/rules/no-identical-names.js
+++ b/lib/rules/no-identical-names.js
@@ -14,6 +14,10 @@ const utils = require("../utils");
 // Rule Definition
 //------------------------------------------------------------------------------
 
+function findLast(arr, callback) {
+    return arr.slice().reverse().find(callback);
+}
+
 module.exports = {
     meta: {
         type: "suggestion",
@@ -51,7 +55,7 @@ module.exports = {
             const parentModule = mapModuleNodeToInfo.get(modulesStack[modulesStack.length - 1]);
             if (parentModule.modules.length > 0) {
                 // Find the last function-less module at the current level if one exists, i.e: module('foo');
-                const lastFunctionLessModule = parentModule.modules.reverse().find(node => node.arguments.length === 1);
+                const lastFunctionLessModule = findLast(parentModule.modules, node => node.arguments.length === 1);
                 if (lastFunctionLessModule) {
                     return lastFunctionLessModule;
                 }

--- a/tests/lib/rules/no-identical-names.js
+++ b/tests/lib/rules/no-identical-names.js
@@ -41,6 +41,17 @@ ruleTester.run("no-identical-title", rule, {
             test("it1", function() {});
         `,
         outdent`
+            module('A');
+
+            test('foo', function() {});
+            test('bar', function() {});
+
+            module('B');
+
+            test('foo', function() {});
+            test('bar', function() {});
+        `,
+        outdent`
             module("module1");
             module("module2");
         `,

